workers/unity/Packages/io.improbable.gdk.buildsystem/WorkerBuilder.cs

```diff
@@ -75,70 +60,19 @@ public static void Build()
-            var result = new List<BuildContext>();
-
-            foreach (var workerType in wantedWorkerTypes)
-            {
-                var environmentConfig =
-                    spatialOSBuildConfiguration.GetEnvironmentConfigForWorker(workerType, buildEnvironment);
-
-                // Filter targets for CI
-                var targetConfigs = environmentConfig.BuildTargets
-                    .Where(t => t.Enabled && (buildTargetFilter?.Contains(t.Target) ?? true))
-                    .ToList();
-
-                // Which build targets are not supported by current install?
-                var missingTargets = targetConfigs
-                    .Where(c => !WorkerBuildData.BuildTargetsThatCanBeBuilt.ContainsKey(c.Target))
-                    .ToList();
-
-                // Error on missing required build support
-                if (missingTargets.Any(c => c.Required))
-                {
-                    var targetNames = string.Join(", ", missingTargets
-                        .Where(c => c.Required)
-                        .Select(c => c.Target.ToString()));
-                    throw new BuildFailedException(
-                        $"Build failed for {workerType}. Cannot build for required ({targetNames}) because build support is not installed in the Unity Editor.");
-                }
-
-                // Log builds we're skipping
-                if (missingTargets.Count > 0)
-                {
-                    var targetNames = string.Join(", ", missingTargets.Select(c => c.Target.ToString()));
-                    Debug.LogWarning(
-                        $"Skipping ({targetNames}) because build support is not installed in the Unity Editor and the build target is not marked as 'Required'.");
-                }
-
-                result.AddRange(targetConfigs.Select(targetConfig => new BuildContext
-                {
-                    WorkerType = workerType, BuildEnvironment = buildEnvironment,
-                    ScriptingImplementation = scriptImplementation ?? ScriptingImplementation.Mono2x,
-                    BuildTargetConfig = targetConfig
-                }));
-            }
-
-            return result;
-        }
-
         internal static void MenuBuild(BuildEnvironment environment, params string[] workerTypes)
         {
             // Delaying build by a frame to ensure the editor has re-rendered the UI to avoid odd glitches.
             EditorApplication.delayCall += () =>
             {
                 try
                 {
-                    // TODO READ FROM SETTINGS
-                    var buildContexts = GetBuildContexts(workerTypes, environment, null, null);
+                    var buildContexts = BuildContext.GetBuildContexts(workerTypes, environment);
                     BuildWorkers(buildContexts);
                 }
-                catch (Exception)
+                catch (Exception e)
                 {
+                    Debug.LogException(e);
```

Comment:
This exception will be logged twice right? 

Here and then in the top-level handler which will unwrap the `BuildFailedException`

Reply:
I kept this the same as it is with the commandline build

Reply:
That's all well and good, but will this be logged twice? 😛

Reply:
I havn't been able to trigger it to log twice. So... I think not?

Reply:
Cool - sgtm